src/test/clojurescript/clara/test_rules_require.cljs

```diff
@@ -0,0 +1,63 @@
+(ns clara.test-rules-require
+  (:require-macros [cljs.test :refer (is deftest run-tests testing)]
+                   [clara.test-rules-data])
+  (:require [cljs.test :as t]
+            [clara.rules.engine :as eng]
+            [clara.rules.accumulators :as acc]
+            [clara.rules :refer [insert retract fire-rules query insert!]
+             :refer-macros [defrule defsession defquery]]
+            [clara.rule-defs :as rd]
+            [clara.rules.testfacts :as facts]))
+
+(comment
+  ;; Launch browser repl.
+  (cemerick.piggieback/cljs-repl :repl-env (cemerick.austin/exec-env))
+  )
+
+(defn- has-fact? [token fact]
+       (some #{fact} (map first (:matches token))))
+
+(defsession my-session 'clara.rule-defs)
+(defsession my-session-map 'clara.rule-defs :fact-type-fn :type)
+
+(deftest test-simple-defrule
+         (let [session (insert my-session (facts/->Temperature 10 "MCI"))]
+
+              (fire-rules session)
+
+              (is (has-fact? @rd/simple-defrule-side-effect (facts/->Temperature 10 "MCI")))
+              (is (has-fact? @rd/other-defrule-side-effect (facts/->Temperature 10 "MCI")))))
+
+(deftest test-simple-query
+         (let [session (-> my-session
+                           (insert (facts/->Temperature 15 "MCI"))
+                           (insert (facts/->Temperature 10 "MCI"))
+                           (insert (facts/->Temperature 80 "MCI"))
+                           fire-rules)]
+
+              ;; The query should identify all items that wer einserted and matchd the
+              ;; expected criteria.
+              (is (= #{{:?t 15} {:?t 10}}
+                     (set (query session rd/cold-query))))))
+
+(deftest test-simple-accumulator
+         (let [session (-> my-session
+                           (insert (facts/->Temperature 15 "MCI"))
+                           (insert (facts/->Temperature 10 "MCI"))
+                           (insert (facts/->Temperature 80 "MCI"))
+                           fire-rules)]
+
+              ;; Accumulator returns the lowest value.
+              (is (= #{{:?t 10}}
+                     (set (query session rd/coldest-query))))))
+
+(deftest test-simple-insert
+         (let [session (-> my-session
+                           (insert (facts/->Temperature 15 "MCI"))
+                           (insert (facts/->WindSpeed 45 "MCI"))
+                           (fire-rules))]
+
+              (is (= #{{:?fact (facts/->ColdAndWindy 15 45)}}
+                     (set
+                       (query session rd/find-cold-and-windy))))))
+
```

Comment:
I don't want to block on this, particularly since this is a problem specific to ClojureScript, but when functionality isn't platform-specific, even if it came up on one platform (JVM vs JS) the use of [def-rules-test](https://github.com/cerner/clara-rules/blob/0.16.1/src/main/clojure/clara/tools/testing_utils.cljc#L17) to create shared tests between Clojure and ClojureScript is preferred.  There are a variety of examples under [test-common](https://github.com/cerner/clara-rules/tree/0.16.1/src/test/common/clara).  I've been gradually working to get the test suite as cross-platform as possible, which has the primary effect of improving test coverage in ClojureScript.  See [this project board](https://github.com/cerner/clara-rules/projects/1).  Not critical to merging this, but a good thing to keep in mind for any future contributions.   The "contributors" readme should probably mention this transition.

Reply:
I'll convert these tests after I get some experience updating the tests on #361.

Reply:
Hmm.. so looking at this again it seems like using def-rules-test would defeat the point of testing the real path of verifying how defrule/defquery work with namespace qualification.  I don't believe I was thinking clearly/paying enough attention there.  Perhaps these could just be cljc tests in the common directory, but look more or less the same as now, just with any necessary changes to the ns header?

Reply:
Moved these tests to common as you suggested, but interestingly the clj version fails. Need to dig in.